File: backends/ruby-hanami/app/domain/client/enums.rb
# frozen_string_literal: true
#
# クライアントのステータス定数を定義するドメインオブジェクトモジュール。
#
# @author Satoshi Nagashiba <satoshi.nagashiba@gmail.com>

module Domain
  module Client
    # クライアントのステータス定数です。
    module Status
      INACTIVE  = 1
      ACTIVE    = 2
      SUSPENDED = 3
      CLOSED    = 4
    end
  end
end

File: backends/kotlin-ktor/src/main/kotlin/com/authorization/domain/client/Enums.kt
/*
 * クライアントドメイン列挙値モジュール。
 *
 * @author Satoshi Nagashiba <satoshi.nagashiba@gmail.com>
 */
package com.authorization.domain.client

/**
 * クライアントステータスの定数です。
 *
 * @author Satoshi Nagashiba <satoshi.nagashiba@gmail.com>
 */
object ClientStatus {
    /** 無効 */
    const val INACTIVE:  Int = 1
    /** 有効 */
    const val ACTIVE:    Int = 2
    /** 停止中 */
    const val SUSPENDED: Int = 3
    /** 解約済み */
    const val CLOSED:    Int = 4
}
